### DISCUSSION TITLE
Use-Cases / Feature Requests

### DISCUSSION THREAD

**Discussion by robertlong** [omigroup/omi-scripting-group#13] [General]
In this thread we can post the use-cases we have in mind for our respective platforms and feature requests for things we'd like to see.

**Comment by robertlong** [comment]
For Third Room I'd like to have a general scene-graph API. You will be able to set the environment for a room and that environment can have a bundled script that can control the whole environment's scene graph. It might have elevated privileges in that it can create / spawn additional objects, has no boundaries, increased resource limits, etc. Objects / avatars that are embedded in the environment may have their own scripts and control their own scene graph. However, these may have a reduced set of permissions, such that an object / avatar cannot expand to fill the entire room, or over-allocate system resources, etc.

**Reply by AdamFrisby** [reply]
I'd second this as mirroring our own expectations of the API as well.

**Comment by technobaboo** [comment]
https://stardustxr.org/docs/stardust-protocol/protocol something similar to the Stardust protocol (but remade to fit WASM) that includes stuff like input handlers, zones, item acceptors, etc. SDFs are insanely important for XR interaction and work very well with 2D interaction as well.

**Comment by avaer** [comment]
Is this topic more general than WASM?

What we are discussing here seems a general API that could work just as well with Javascript rather than WASM (that might be a more natural place to start?). Anything that works in JS usually has a natural binding to WASM.

**Reply by RangerMauve** [reply]
I think it'd be nice to focus on WASM for situations where you're running in environments without JavaScript (e.g. Unity/Godot/Unreal). Like, if the API could be adopted to non-WASM, that'd be useful, but a WASM-first rather than a JS-first approach will probably make integration easier outside the browser.
